spec/chrome_integration/gtalk.spec.js

```diff
@@ -0,0 +1,237 @@
+          clientState.status == 'ONLINE') {
+        fulfill(clientState);
+      }
+    });
+  });
+}
+
+describe('GTalk', function() {
+  var socialInterface;
+
+  // Message to be sent between peers.  If a unique message is not used,
+  // messages from one persons test might interfere with another person who
+  // is running the same tests at the same time.
+  var uniqueMsg = Math.random().toString();
+
+  function logoutThenDone(socialClientsArray, done) {
+    var promises = [];
+    for (var i = 0; i < socialClientsArray.length; ++i) {
+      promises.push(socialClientsArray[i].logout());
+    }
+    Promise.all(promises).then(done);
+  }
+
+  beforeEach(function(done) {
+    if (!socialInterface) {
+      freedom('scripts/dist/social.google.json',
+          {oauth: [OAuthView], debug: 'log'})
+          .then(function(interface) {
+        // Store socialInterface so we don't have to reload freedom
+        // before each test.
+        socialInterface = interface;
+        done();
+      }.bind(this));
+    } else {
+      done();
+    }
+  }.bind(this));
+
+  it('Can login and logout', function(done) {
+    var socialClient = socialInterface();
+    loginAs(socialClient, ALICE_EMAIL).then(function(clientInfo) {
+      expect(clientInfo.userId).toEqual(ALICE_EMAIL);
+      expect(clientInfo.clientId).toEqual(ALICE_EMAIL + '/integration');
+      expect(clientInfo.status).toEqual('ONLINE');
+      logoutThenDone([socialClient], done);
+    });
+  });
+
+  it('Can send messages', function(done) {
+    // 1st login as Alice.
+    var aliceSocialClient = socialInterface();
+    loginAs(aliceSocialClient, ALICE_EMAIL).then(function(aliceClientInfo) {
+      // Next setup a listener to send Bob messages when he is online
+      waitForUser(aliceSocialClient, BOB_ANONYMIZED_ID)
+          .then(function(clientState) {
+        aliceSocialClient.sendMessage(clientState.clientId, uniqueMsg);
+      });
+
+      // Next login as Bob and monitor for messages.
+      var bobSocialClient = socialInterface();
+      loginAs(bobSocialClient, BOB_EMAIL).then(function(bobClientInfo) {
+        bobSocialClient.on('onMessage', function(messageData) {
+          if (messageData.from.userId == ALICE_ANONYMIZED_ID &&
+              messageData.message == uniqueMsg) {
+            logoutThenDone([aliceSocialClient, bobSocialClient], done);
+          }
+        });  // end of bobSocialClient.on('onMessage', ...
+      });  // end of loginAs(bob...)
+    });  // end of loginAs(alice...)
+  });
+
+  // We should be able to send 8 messages per second from one peer to another
+  // without being throttled by GTalk (i.e. we should not get 503 "service
+  // unavailable" errors).
+  it('Can send 8 messages per second', function(done) {
+    var NUM_MESSAGES_PER_SECOND = 8;
+    // Defined in socialprovider.js
+    // TODO: make this a constant available to users of the freedom social
+    // provider object
+    var BATCH_INTERVAL_MS = 100;
```

Comment:
Just used to calculate MESSAGE_FREQUENCY.  I could remove it and just do ```var MESSAGE_FREQUENCY = 110;``` but I thought having this extra constant is more clear